kuma/javascript/src/bcd-signal.jsx

```diff
@@ -0,0 +1,809 @@
+// @flow
+import { gettext, interpolate } from './l10n.js';
+import { getCookie } from './utils.js';
+
+let bcSignalsRendered = false;
+
+export function activateBCDSignals(slug: string, locale: string) {
+    if (bcSignalsRendered) {
+        return;
+    }
+
+    let bcTable;
+    let bcSignalBlock;
+    let errorMessageWrapper;
+    let bcSignalCompleteBlock;
+    let sendReportButton;
+    let nextStepButton;
+    let stepsInfoSpan;
+    let bcSignalStep;
+    let bcSignalSteps = 2;
+
+    /**
+     * Toggles main block visibility and resets form completion progress
+     */
+    const toggleBcSignalBlock = () => {
+        if (bcSignalCompleteBlock.classList.contains('open')) {
+            bcSignalCompleteBlock.classList.remove('open');
+        } else {
+            bcSignalBlock.classList.toggle('open');
+        }
+
+        resetControls();
+        toStep(1);
+    };
+
+    /**
+     * Moves user to a step, by settings displaying related blocks
+     * @param {Number} step - Step index to be shown
+     */
+    const toStep = step => {
+        if (step !== bcSignalStep) {
+            const prevStep = bcSignalStep;
+            bcSignalStep = step;
+            stepsInfoSpan.innerText = interpolate(
+                gettext('Step %(current)s of %(total)s'),
+                {
+                    current: bcSignalStep,
+                    total: bcSignalSteps
+                }
+            );
+
+            const stepId =
+                document && document.getElementById(`step-${bcSignalStep}`);
+            if (stepId && stepId.classList) {
+                stepId.classList.add('active');
+            }
+
+            const prevStepId =
+                document && document.getElementById(`step-${prevStep}`);
+            if (prevStep && prevStepId && prevStepId.classList) {
+                prevStepId.classList.remove('active');
+            }
+        }
+    };
+
+    /**
+     * Resets all the controls to initial state
+     */
+    const resetControls = () => {
+        for (const browser of document.querySelectorAll('.browser.selected')) {
+            browser.classList.remove('selected');
+        }
+
+        const brief = document && document.getElementById('brief-explanation');
```

Comment:
The condition is suggested by ci-flow. Flow requires explicit type checks, otherwise the code doesn't pass the CircleCI flow tests. There are other occurrences of this kind of checks, all for the same reason.

Reply:
For the record, @Gregoor suggested your can solve it by defining a type. See https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoAJgUwMYwIYBOmYAboWOnNgK4C2mAdgC4BcYA3tkXk5gKIxM9ZmwAUAZzbimBAJYMA5gEowAXgB8HXHnHiAMrOks8DAJ4AaMNt0A5PPTYnTAX2cBuVKkFMwmAgTgCAFlMXTwFTAB1AjwAB1i-D1RsOAZpK25ePn9AkLCIgAkTdEECNTBRFQ0OVDBfHODQ8XComPi-csoaYSYAOi5MHn5BHtEAcnRZEjGlDzq-AMb81riEgn78W3tiVTAxhEIGeQV6xYBaemWxj3dPFLSfInFMJgBhVJk4GHFyyrVNdi1U65JotaKrPwbHT6Qx9Ii0OAkTDjEiGWQAI0EMxuSQGQ2yizyzUKxVKlQ8Txe72YAW+5KAA
I think it can wait because perhaps a better solution altogether is to rewrite everything in React but that too can wait.

Reply:
@peterbe thanks for your note. We'll keep this for the record as you said, and wait with the implementation until we decide that we really need this implemented, or everything is rewritten in React at a later date.